cmake/std/trilinosprhelpers/TrilinosPRConfigurationBase.py

```diff
@@ -0,0 +1,587 @@
+        if self._trilinos_pr_env is None:
+            self._trilinos_pr_env = jenkinsenv.TrilinosJenkinsEnv()
+        return self._trilinos_pr_env
+
+
+    @property
+    def config_data(self):
+        """
+        Configuration data.
+
+        This is a setenvironment.SetEnvironment class instance containing the information
+        loaded from the config.ini file.
+        """
+        if self._config_data is None:
+            self._config_data = setenvironment.SetEnvironment(self.args.configfile,
+                                                              self.arg_pr_jenkins_job_name)
+        return self._config_data
+
+
+    @property
+    def config_script(self):
+        """
+        Returns the configuration script from the configuration file.
+
+        This function searches the [CONFIG_SCRIPT_MAP] section in the config.ini file
+        and looks for the key matching the Jenkins job name and returns the
+        value from that KV pair.  This would be the .cmake file that maps to the
+        job.
+
+        Returns:
+            String containing the job-specific configuration script to load.
+        """
+        return self.get_property_from_config("CONFIG_SCRIPT_MAP", self.arg_pr_jenkins_job_name)
+
+
+    @property
+    def max_cores_allowed(self):
+        """
+        Calculate the true max number of cores we can use.
+
+        This is the minimum of the available cores detected on the system
+        and the max cores allowed by argument.
+
+        If the max_cores_allowd is not > 0 then we use the # of cores on the system.
+        """
+        if self._max_cores_allowed is None:
+            num_cores = multiprocessing.cpu_count()
+            if self.arg_max_cores_allowed > 0:
+                self._max_cores_allowed = min(num_cores, int(self.arg_max_cores_allowed))
+            else:
+                self._max_cores_allowed = num_cores
+        return self._max_cores_allowed
+
+
+    @property
+    def max_test_parallelism(self):
+        """
+        Maximum parallelism of your tests. This is obtained from the configuration.ini
+        file from the [PR_JOB_PARAMETERS] section key max-test-parallelism value.
+        If this value is missing from the configuration file, we default to 1.
+        """
+        if self._max_test_parallelism is None:
+            try:
+                self._max_test_parallelism = int(self.get_property_from_config("PR_JOB_PARAMETERS","max-test-parallelism"))
+            except:
+                self._max_test_parallelism = 1
+        return self._max_test_parallelism
+
+
+    @property
+    def concurrency_build(self):
+        """
+        Concurrency to use for building Trilinos
+
+        This is equvalent to running `make -j <concurrency_build>` from the command line.
+        """
+        if self._concurrency_build is None:
+            si = sysinfo.SysInfo()
+
+            self._concurrency_build = si.compute_num_usable_cores(req_mem_gb_per_core = self.arg_req_mem_per_core,
```

Comment:
Should there be spaces here:  `req_mem_gb_per_core = self.arg_req_mem_per_core`?

Reply:
To me, this:
```python
            self._concurrency_build = si.compute_num_usable_cores(req_mem_gb_per_core = self.arg_req_mem_per_core,
                                                                  max_cores_allowed   = self.max_cores_allowed)
```

is easier to read than:

```python
            self._concurrency_build = si.compute_num_usable_cores(req_mem_gb_per_core=self.arg_req_mem_per_core,
                                                                  max_cores_allowed=self.max_cores_allowed)
```

I like whitespace and I can not lie...

Reply:
> I like whitespace and I can not lie...

Oh, I heartily agree.  I thought the spaces were a requirement, but maybe that's just a PEP8 thing.

Reply:
pep8 hates whitespace - I never got that...